lib/spack/spack/fetch_strategy.py

```diff
@@ -392,7 +392,26 @@ def __init__(self, *args, **kwargs):
 
     @_needs_stage
     def fetch(self):
-        super(CacheURLFetchStrategy, self).fetch()
+        path = re.sub('^file://', '', self.url)
```

Comment:
This one was also designed to be safe -- I think it should take a path or a URL.